tools/pscript/cli.c

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (C) 2017, Feng Liu
+ **/
+#include <stdio.h>
+#include <signal.h>
+#include <readline/readline.h>
+#include <readline/history.h>
+
+#include <utils/log.h>
+
+#include <pss.h>
+
+#include <plumber.h>
+#include <builtin.h>
+#include <cli.h>
+
+extern pss_vm_t* current_vm;
+
+static const char* _prompt = "PSS> ";
+
+static int _interrupt = 0;
+
+struct _line_list_t {
+	char *line;
+	uint32_t off;
+	uint32_t size;
+	struct _line_list_t *next;
+};
+
+/** @brief concatenate the lines in line list
+ */
+static char* _cat_lines(struct _line_list_t *head, uint32_t code_size)
```

Comment:
code_size is the total size of the code buffer, I need this code_size to malloc memory.